metadata-ingestion/src/datahub/ingestion/source/datahub/config.py

```diff
@@ -0,0 +1,52 @@
+from typing import Optional
+
+from pydantic import Field
+
+from datahub.configuration.kafka import KafkaConsumerConnectionConfig
+from datahub.ingestion.source.sql.mysql import MySQLConfig
+from datahub.ingestion.source.state.stateful_ingestion_base import (
+    StatefulIngestionConfig,
+    StatefulIngestionConfigBase,
+)
+
+
+class DataHubSourceConfig(StatefulIngestionConfigBase):
+    mysql_connection: MySQLConfig = Field(
+        # TODO: Check, do these defaults make sense?
+        default=MySQLConfig(username="datahub", password="datahub", database="datahub"),
```

Comment:
MySQLConfig also has table_pattern, view_pattern, domain, etc

seems like we might need something separate here

Overall - we should always have a split of "connection config" vs "source config", where the latter inherits/embeds the former

Reply:
also probably doesn't make sense to have a default here

Reply:
Yup, I'll remove defaults. I guess I can start disentangling our configs here

Reply:
Changes made, which didn't really make things simpler unfortunately:
- Renamed SQLAlchemyConfig -> SQLCommonConfig
- Split out the connection parts (i.e. all) of BasicSQLAlchemyConfig into `SQLAlchemyConnectionConfig`
- Created `MySQLConnectionConfig` off of `SQLAlchemyConnectionConfig` and `MySQLConfig` now also inherits from `MySQLConnectionConfig`